luda-adventure/src/app/game/update/calculate_next_motion_of_character/calculate_next_movement_of_character.rs

```diff
@@ -0,0 +1,279 @@
+use crate::{
+    app::game::{
+        known_id::object::PLAYER_CHARACTER, simplify_collision_box_list, zero_speed, Collider,
+        CollisionBox, MovementPath, PlayerCharacter, Positioner, TileExt, Velocity,
+    },
+    ecs,
+};
+use float_cmp::{ApproxEq, F32Margin};
+use namui::prelude::*;
+
+const COLLISION_MARGIN: F32Margin = F32Margin {
+    ulps: 1,
+    epsilon: 0.0001,
+};
+
+pub fn calculate_next_movement_of_character(ecs_app: &mut ecs::App, max_duration: Time) {
```

Comment:
I'm not sure yet what do you want to do with this. I know the logic but I feel it's very complicated.

Positioner, it seems having two state, Fixed and Dynamic, and most of code is for Dynamic case. I feel this very strange and i'm not clear with it.

Reply:
I think so too. But I couldn't find a better way.
Outdated